docs-src/cloud/terraform-provider.md

```diff
@@ -0,0 +1,164 @@
+---
+id: terraform-provider
+title: Temporal Cloud Terraform provider
+sidebar_label: Terraform provider
+description:
+tags:
+  - temporal cloud
+  - terraform
+ssdi:
+  - The Terraform Provider is in a Public Preview release status for Temporal Cloud.
+---
+
+The Terraform Temporal provider is a plugin that allows Terraform to manage resources on Temporal Cloud.
+Terraform is a tool used to codify and provision infrastructure as code (IaC).
+With this provider, you can use Terraform to automate the management of Temporal Cloud Namespaces, resources for Temporal Cloud users, and more.
+
+The [Temporal Cloud Terraform provider](https://registry.terraform.io/providers/temporalio/temporalcloud/latest) is available in the Terraform Registry, where you can find detailed documentation on the supported resources and data sources.
+
+The GitHub repository for the Terraform provider is [terraform-provider-temporalcloud](https://github.com/temporalio/terraform-provider-temporalcloud/tree/main), where you can find the source code, contribution guidelines, and additional information.
+
+:::note
+
+Temporal supports the [Cloud Namespace](/cloud/namespaces) resource management, a user resource for managing Temporal Cloud users, and data sources for retrieving information about existing Namespaces and users, with the Temporal Cloud Terraform provider.
+
+:::
+
+### Prerequisites
+
+You must have access to the following to use the Temporal Cloud Terraform provider:
+
+- [Terraform account](https://developer.hashicorp.com/terraform)
+- [Terraform CLI](https://developer.hashicorp.com/terraform/cli)
+- [Temporal tcld](/cloud/tcld/)
+- [Issued certification authority](/cloud/certificates)
+
+## Get started
+
+This walk you through getting started with the Temporal Cloud Terraform provider.
+You will generate an API Key used to authenticate your requests with your Temporal Cloud account.
+Then you will create a Temporal Cloud Namespace with the Terraform provider.
+
+### 1. Generate an API Key
+
```

Comment:
```suggestion

**How to generate an API Key.**

```

Use a question like sentence at the top to frame the scope of the information for the LLMs.